pandas/core/reshape/merge.py

```diff
@@ -220,10 +216,10 @@ def merge_ordered(
         left_on docs.
     left_by : column name or list of column names
         Group left DataFrame by group columns and merge piece by piece with
-        right DataFrame.
+        right DataFrame. Must be None if either left or right are a Series.
```

Comment:
I didn't immediately find tests raising if this was the case. Mind adding a unit test to confirm this behavior?